eclair-core/src/main/scala/fr/acinq/eclair/UInt64.scala

```diff
@@ -16,40 +16,43 @@
 
 package fr.acinq.eclair
 
-import java.math.BigInteger
-
+import com.google.common.primitives.UnsignedLongs
 import scodec.bits.ByteVector
+import scodec.bits.HexStringSyntax
 
-case class UInt64(private val underlying: BigInt) extends Ordered[UInt64] {
+case class UInt64(private val underlying: Long) extends Ordered[UInt64] {
 
-  require(underlying >= 0, s"uint64 must be positive (actual=$underlying)")
-  require(underlying <= UInt64.MaxValueBigInt, s"uint64 must be < 2^64 -1 (actual=$underlying)")
+  override def compare(o: UInt64): Int = UnsignedLongs.compare(underlying, o.underlying)
+  private def compare(other: MilliSatoshi): Int = other.toLong match {
+    case l if l < 0 => 1                          // if @param 'other' is negative then is always smaller than 'this'
+    case _ => compare(UInt64(other.toLong))
+  }
 
-  override def compare(o: UInt64): Int = underlying.compare(o.underlying)
+  def <(other: MilliSatoshi): Boolean = compare(other) < 0
+  def >(other: MilliSatoshi): Boolean = compare(other) > 0
+  def <=(other: MilliSatoshi): Boolean = compare(other) <= 0
+  def >=(other: MilliSatoshi): Boolean = compare(other) >= 0
 
-  def toByteVector: ByteVector = ByteVector.view(underlying.toByteArray.takeRight(8))
+  def toByteVector: ByteVector = underlying match {
+    case 0 => hex"00"
+    case _ => ByteVector.fromLong(underlying).dropWhile(_ == 0)
```

Comment:
The `dropWhile` shouldn't be necessary anymore

Reply:
I think we need it, otherwise we will always have something that's 8-bytes long?

Reply:
Replacing the whole method by:

```scala
  def toByteVector: ByteVector = ByteVector.fromLong(underlying)
```

should work (except that it makes cosmetic tests fail). I think it is simpler that way, do you see a potential problem?